Doc: Add section for resetting cluster state that is stuck (#217)
azimuth-cloud/azimuth-config: docs/debugging/kubernetes.md

kubectl -n capi-addon-system logs deploy/cluster-api-addon-provider
```

### Recovering clusters stuck in failed state after network disruption

> If Azimuth suffered networking problem, clusters can get stuck in 'Failed' state

Reviewer: ```suggestion
If Azimuth has suffered networking problems, clusters can get stuck in a 'Failed' state
```

> even though the network is recovered and all the pods are fine.

Reviewer: ```suggestion
even after the network has recovered and the cluster is otherwise fully functional.
```

> This is can happend when ``failureMessage`` and ``failureReason`` are not

Reviewer: ```suggestion
This is can happen when ``failureMessage`` and ``failureReason`` are set, which mistakenly indicates an unrecoverable error. These must be manually
```

cleared.

> If you think this is your case, you can check clusters that are affected by following command.

Reviewer: ```suggestion
If you think this is the case, you can check for affected clusters with the following command:
```

```command  title="On the K3s node, targetting the HA cluster if deployed"
$ kubectl get cluster.cluster.x-k8s.io --all-namespaces -o json | jq -r '.items[] | "\(.metadata.name): \(.status.failureMessage) \(.status.failureReason)"'
```

> Clusters with either one of field (or both) is not ``null`` are affected ones.

Reviewer: ```suggestion
Clusters where either field is not ``null`` are affected.
```

> Clusters where either one of field is not ``null`` are affected.

Reviewer 2: ```suggestion
Clusters where one or both of the `failure{Message,Reason}` fields is not `null` are affected.
```

> You can reset their status by updating these values to ``null`` using
> ``kubectl patch``

Reviewer 2: ```suggestion
You can reset the status for an individual cluster by updating removing the failure message and reason fields using `kubectl edit --subresource=status clusters.cluster.x-k8s.io/<cluster-name>`. Alternatively, you can apply a patch to all workload clusters at once using the following command:
```

```command  title="On the K3s node, targetting the HA cluster if deployed"

> ##### TEST IF COMMANDS ARE CRAFTED CORRECTLY BEFORE ACTUAL PATCH

Reviewer 2: It took me longer than it should have to realise what this wall of text was actually doing 😄 so probably best to explain it more clearly in the comment
```suggestion
# Shell command to extract the list of failed clusters and generate the required `kubectl patch` command for each one
```

$ kubectl get cluster.cluster.x-k8s.io --all-namespaces -o json \
| jq -r '.items[] | select(.status.failureMessage or .status.failureReason) | "kubectl patch cluster.cluster.x-k8s.io \(.metadata.name) -n \(.metadata.namespace) --type=merge --subresource=status -p '\''{\"status\": {\"failureMessage\": null, \"failureReason\": null}}'\''"'
kubectl patch cluster.cluster.x-k8s.io demo1 -n az-demo --type=merge --subresource=status -p '{"status": {"failureMessage": null, "failureReason": null}}'
kubectl patch cluster.cluster.x-k8s.io demo2 -n az-demo --type=merge --subresource=status -p '{"status": {"failureMessage": null, "failureReason": null}}'
kubectl patch cluster.cluster.x-k8s.io demo3 -n az-demo --type=merge --subresource=status -p '{"status": {"failureMessage": null, "failureReason": null}}'
kubectl patch cluster.cluster.x-k8s.io demo4 -n az-demo --type=merge --subresource=status -p '{"status": {"failureMessage": null, "failureReason": null}}'

> ##### FIX FAILED CLUSTERS (ADD '| sh' AT THE END TO APPLY PATCHES)

Reviewer 2: ```suggestion
# Modification of the previous command which pipes the output into `sh` so that the `kubectl patch` commands are executed to fix the failed clusters
```
